### PR TITLE
[JENKINS-45875] Bump AHC dependency to 2.0.33

### PR DESCRIPTION
**Pull request by ghost**
@reviewbybees @stephenc 

https://issues.jenkins-ci.org/browse/JENKINS-45875

### CODE REVIEW

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

File: pom.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
   </parent>
 
   <artifactId>async-http-client</artifactId>
```
Comment (stephenc) [inline pom.xml]: <artifactId>async-http-client2</artifactId> as version 2 is not binary compatible

**Review by stephenc — CHANGES_REQUESTED** [review]
Breaks binary compatibility and will blow up any plugins that depend on the previous version. We just need a new plugin for the new 2.x series with classes moved to a new package too

**Comment by ghost** [comment]
@stephenc done

**Review by stephenc — APPROVED** [review]

**Comment by stephenc** [comment]
Only question is whether we shouldn't just change to a new repo now that we are changing artifact id? @daniel-beck

**Comment by daniel-beck** [comment]
@stephenc No opinion. As this is basically just a major version bump, and taking both Jenkins limitations and the type of plugin into account, this seems reasonable. More reasonable than either a copy of this repo, or starting over without history (assuming there's useful history), while this repo dies out.

Maybe @oleg-nenashev @orrc @batmat have an opinion.

File: pom.xml [hunk 2]
```diff
@@ -63,7 +63,7 @@
 
   <properties>
     <jenkins.version>1.580.1</jenkins.version>
```
Comment (oleg-nenashev) [inline pom.xml:65]: 🐛 Jenkins version should be changed

**Review by oleg-nenashev — COMMENTED** [review]
@stephenc So do you plan to have 2 independent release lines in the same repo? Maybe it worth to just split them to separate repos for better readability

**Comment by ghost** [comment]
@oleg-nenashev @daniel-beck @stephenc considering closing this in favor of #8, as there are different issues that would need to be sorted out for the major version bump that don't seem worth the time.

**Comment by recampbell** [comment]
I guess we should test installing both async-http-client.jpi and async-http-client2.jpi on the same instance to make sure you can have both installed at once.

**Comment by recampbell** [comment]
I think we should definitely do #8, but if we also had this available and recommended, then plugins could start moving to it on their own time, right?

**Comment by oleg-nenashev** [comment]
@recampbell I think we need both, but IMHO this PR should become a separate plugin in a separate repository. It should be safe to use both plugins in parallel from what I see

**Pull request closed without merging**